### PR TITLE
KeyVault: Updated roles

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Updated RBAC roles to latest
 
Pipeline reference
[KeyVault: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.KeyVault/vaults/deploy.bicep [hunk 1]
```diff
@@ -179,22 +179,22 @@ resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' = {
     softDeleteRetentionInDays: softDeleteRetentionInDays
     enableRbacAuthorization: enableRbacAuthorization
     createMode: createMode
-    enablePurgeProtection: ((!enablePurgeProtection) ? null : enablePurgeProtection)
+    enablePurgeProtection: enablePurgeProtection ? enablePurgeProtection : null
```
Comment (MariusStorhaug) [inline arm/Microsoft.KeyVault/vaults/deploy.bicep:182]:
```suggestion
    enablePurgeProtection: enablePurgeProtection
```
Is this not a boolean parameter with a default value of `false`? Why are we setting it to null?

File: arm/Microsoft.KeyVault/vaults/deploy.bicep [hunk 2]
```diff
@@ -179,22 +179,22 @@ resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' = {
...
     tenantId: subscription().tenantId
     accessPolicies: formattedAccessPolicies
     sku: {
       name: vaultSku
       family: 'A'
     }
-    networkAcls: (empty(networkAcls) ? null : networkAcls_var)
+    networkAcls: !empty(networkAcls) ? networkAcls_var : null
```
Comment (MariusStorhaug) [inline arm/Microsoft.KeyVault/vaults/deploy.bicep:189]: Can you clean up braces on the variable `networkAcls_var`?

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]

File: arm/Microsoft.KeyVault/vaults/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/deploy.bicep:182]: @MariusStorhaug we actually had an issue for that (https://github.com/Azure/ResourceModules/issues/1038#issuecomment-1057134138). In essence, that service only accepts `true` or nothing.

File: arm/Microsoft.KeyVault/vaults/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.KeyVault/vaults/deploy.bicep:189]: done

File: arm/Microsoft.KeyVault/vaults/deploy.bicep (hunk 1 above)
Comment (MariusStorhaug) [reply arm/Microsoft.KeyVault/vaults/deploy.bicep]: :D

Comment (MariusStorhaug) [reply arm/Microsoft.KeyVault/vaults/deploy.bicep]: Should we raise that to the PG?

**Review by MariusStorhaug — APPROVED** [review]

**Pull request merged**